### DISCUSSION TITLE
Tabbing Order - when answering questions

### DISCUSSION THREAD

**Discussion by MajorMajorMaj** [openwebwork/webwork2#2459] [General]
The current tabbing order is being disrupted by MathQuill. When moving from one input field to another, "Tab" moves directly to the mathquill toolbar instead of the next input field. 
<img width="620" alt="Current order of elements visual" src="https://github.com/user-attachments/assets/f45db973-3d1d-4551-a4ba-abcc42f633f4">
**the orange arrow is the 1st tab action, red 2nd etc...**

I believe it should work like this
![image](https://github.com/user-attachments/assets/e43c14f2-f24c-4818-877e-dc0b3e42afbf)

I understand the want for the toolbar to be tabbable, but most of the time the user is tabbing from answer to answer. 
There seems to be an instance of the toolbar with every inputtable element. This means that the tab order will always go from input element to toolbar instead of input element to input element. Also, I might add that even parts of the answer (non input elements) are tabbable, an example being "0", "A1", "A2", "A3" etc... in the example below
![image](https://github.com/user-attachments/assets/3d62ec4c-2133-4b10-8d85-114483a370e2)

I'm not sure if this a webwork-wide problem, or just the implementation at my school which is currently using v2.18 (I'm a student who wants to be able to be able to turn in his homework faster).

Thanks for reading, hope this explanation is sufficient.

**Comment by drgrice1** [comment]
Unfortunately, for accessibility purposes the tab order as it currently is is correct.  It can not be done the way you suggest.  Note that you can disable the toolbar by right clicking on a MathQuill input and clicking on "Disable Toolbar".  It can be re-enabled again in much the same way.  If the toolbar is enabled, you can also type "Esc" after tabbing into the toolbar, and the toolbar will close and focus will move on to the next element.

Also, the reason the non-input elements are in the tab order is because those are MathJax elements.  Those also must be in the tab order for accessibility purposes.  You can disable that by right clicking on one of them and in the "Accessibility" sub menu uncheck "Include in tab order"

**Comment by Alex-Jordan** [comment]
I will double check, but I believe the new student orientation with 2.19
that comes with the distribution describes how to disable the toolbar for
this reason. But probably does not mention disabling tabbing on MathJax
items. I can add that. All we can do is inform users about these things
because as Glenn says, they need to be the way they are for general
accessibility.

**Comment by MajorMajorMaj** [comment]
Thanks for the responses and the tip about right clicking the input field to disable mathquill toolbar. I think it's worth mentioning from a users perspective that disabling the toolbar in such a way can almost be considered an easter egg (the action lacks user intuition, and most people don't read user manuals or go through orientation). 

I would love to have a positive impact. I will learn more and try to find a solution that allows a decent tabbing experience and doesn't comprise on accessibility.
If there are specific accessibility guidelines or requirements I would appreciate if you can point me in the right direction. Thanks for your time, have a great day.
